conans/test/functional/scm_test.py

```diff
@@ -476,6 +476,70 @@ def scm_serialization_test(self):
         data2 = json.loads(the_json)
         self.assertEquals(data, data2)
 
+    def test_git_delegated_function(self):
+        conanfile = """
+import os
+from conans import ConanFile
+from conans.client.tools.scm import Git
+
+def get_revision():
+    here = os.path.dirname(__file__)
+    git = Git(here)
+    return git.get_commit()
+    
+def get_url():
+    def nested_url():
```

Comment:
Just to check that functions are fully evaluated. My first thought was that nested functions were not being evaluated (no sense), now I'm sure that they are adding nothing to the test, but I will leave it because I need a test evaluating functions.